folding-schemes/src/commitment/kzg.rs

```diff
@@ -70,19 +81,44 @@ impl<'a, C: CurveGroup> Valid for ProverKey<'a, C> {
     }
 }
 
+/// Proof of polynomial evaluation at a point.
+///
+/// Contains both the claimed evaluation and a KZG proof element.
 #[derive(Debug, Clone, Default, Eq, PartialEq, CanonicalSerialize, CanonicalDeserialize)]
 pub struct Proof<C: CurveGroup> {
+    /// The claimed evaluation value f(z)
     pub eval: C::ScalarField,
+
+    /// The proof element π = (f(X) - f(z))/(X - z)
     pub proof: C,
 }
 
-/// KZG implements the CommitmentScheme trait for the KZG commitment scheme.
+/// KZG polynomial commitment scheme implementation.
+///
+/// This implements the [`CommitmentScheme`] trait for KZG polynomial commitments.
+/// The type parameter `H` controls whether hiding commitments are used (currently unsupported).
+///
+/// # Type Parameters
+///
+/// * `'a` - Lifetime of the prover parameters
+/// * `E` - The pairing engine
+/// * `H` - Whether hiding commitments are used (must be false currently)
 #[derive(Debug, Clone, Default, Eq, PartialEq)]
 pub struct KZG<'a, E: Pairing, const H: bool = false> {
+    /// Inner lifetime accounting
     _a: PhantomData<&'a ()>,
+    /// Inner [`Pairing`] friendly curve
     _e: PhantomData<E>,
 }
 
+/*
+TODO (autoparallel): Okay, I'm noticing something here, basically I think that there should likely be two implementations for `CommitmentScheme<G>`,
+one that is hiding, and one that is not (as opposed to the const generic `H` in the trait itself). We could have `HidingCommitmentScheme: CommitmentScheme`.
+If impl `CommitmentScheme<G> for KZG<'a, E, H>`, then we can impl `HidingCommitmentScheme for KZG<'a, E, true>`. The implementation of `HidingCommitmentScheme`
+would be super straight forward as it would just add the blinding factor to the output of the "super" `CommitmentScheme` `commit` and `prove` methods. Then those
+methods on `CommitmentScheme` do not have to take in `blind: Option<E::ScalarField>` or the `dyn Rng`.
```

Comment:
Would this mean having almost the same code twice for each the commitment scheme twice? (one struct for hiding and one struct for non hiding)
With the `H` param we reuse the (almost) same impl while both (hiding &non-hiding) are available.

From the using point of view, since the `H` is set to default `H=false`, if when using the struct it is not specified (just not setting any `H` value), it acts like is `false`.
In other words,
- the `NonHidingCommitmentScheme` currently is achieved by `CommitmentScheme<G1>` (where `H=false` by default)
- and the `HidingCommitmentScheme` is achieved by `CommitmentScheme<G1, true>`